Analysis/Tasks/PWGHF/HFLcK0spCandidateSelector.cxx

```diff
@@ -0,0 +1,336 @@
+      LOG(DEBUG) << "massK0s cut failed: from v0 in cascade, K0s --> " << hfCandCascade.mK0Short() << ", in PDG K0s --> " << RecoDecay::getMassPDG(kK0Short) << ", cut --> " << cuts[pTBin][0];
+      return false; // mass of the K0s
+    }
+
+    if ((std::abs(hfCandCascade.mLambda() - RecoDecay::getMassPDG(kLambda0)) < cuts[pTBin][1]) || (std::abs(hfCandCascade.mAntiLambda() - RecoDecay::getMassPDG(kLambda0)) < cuts[pTBin][1])) {
+      LOG(DEBUG) << "mass L cut failed: from v0 in cascade, Lambda --> " << hfCandCascade.mLambda() << ", AntiLambda --> " << hfCandCascade.mAntiLambda() << ", in PDG, Lambda --> " << RecoDecay::getMassPDG(kLambda0) << ", cut --> " << cuts[pTBin][1];
+      return false; // mass of the Lambda
+    }
+
+    if (std::abs(InvMassGamma(hfCandCascade) - RecoDecay::getMassPDG(kGamma)) < cuts[pTBin][2]) {
+      LOG(DEBUG) << "mass gamma cut failed: from v0 in cascade, gamma --> " << InvMassGamma(hfCandCascade) << ", cut --> " << cuts[pTBin][2];
+      return false; // mass of the Gamma
+    }
+
+    if (hfCandCascade.ptProng0() < cuts[pTBin][3]) {
+      LOG(DEBUG) << "v0 pt cut failed, from cascade --> " << hfCandCascade.ptProng0() << ", cut --> " << cuts[pTBin][3];
+      return false; // pt of the K0
+    }
+
+    if (hfCandCascade.ptProng1() < cuts[pTBin][4]) {
+      LOG(DEBUG) << "bach pt cut failed, from cascade --> " << hfCandCascade.ptProng1() << " , cut --> " << cuts[pTBin][4];
+      return false; // pt of the p
+    }
+
+    if (hfCandCascade.pt() < cuts[pTBin][5]) {
+      LOG(DEBUG) << "cand pt cut failed, from cascade --> " << hfCandCascade.pt() << ", cut --> " << cuts[pTBin][5];
+      return false; // pt of the Lc
+    }
+
+    if (std::abs(hfCandCascade.impactParameter1()) > cuts[pTBin][6]) {
+      LOG(DEBUG) << "d0 bach cut failed, in cascade --> " << hfCandCascade.impactParameter1() << ", cut --> " << cuts[pTBin][6];
+      return false; // d0 of the bachelor
+    }
+
+    /*
+    if ((std::abs(hfCandCascade.dcapostopv()) > cuts[pTBin][7]) || (std::abs(hfCandCascade.dcanegtopv()) > cuts[pTBin][7])) {
+      LOG(DEBUG) << "v0 daugh cut failed, positive v0 daugh --> " << hfCandCascade.dcapostopv() << ", negative v0 daugh --> " << hfCandCascade.dcanegtopv() << " , cut --> " << cuts[pTBin][7];
+      return false; // d0 of the K0s daughters
+    }
+    */
+
+    if (std::abs(hfCandCascade.impactParameter0()) > cuts[pTBin][6]) {
+      LOG(DEBUG) << "d0 v0 cut failed, in cascade --> " << hfCandCascade.impactParameter0() << ", cut --> " << cuts[pTBin][7];
+      return false; // d0 of the v0
+    }
+
+    return true;
+  }
+
+  /// Check if track is ok for TPC PID
+  /// \param track is the track
+  /// \note function to be expanded
+  /// \return true if track is ok for TPC PID
+  template <typename T>
+  bool validTPCPID(const T& track)
+  {
+    if (TMath::Abs(track.pt()) < d_pidTPCMinpT || TMath::Abs(track.pt()) >= d_pidTPCMaxpT) {
```

Comment:
This doesn't make sense to me. Why would `pt` be negative?
@nzardosh This is already in `HFD0CandidateSelector`. Why?

Reply:
I had the same doubt, but thought that this was already thought over

Reply:
It's a bug. Please fix it.

Reply:
done. It is not a bug in the true sense, but clearly it is useless and introduces an overhead.